rosbag2_cpp/CMakeLists.txt

```diff
@@ -33,6 +33,14 @@ if(NOT DISABLE_SANITIZERS AND CMAKE_COMPILER_IS_GNUCXX)
   endif()
 endif()
 
+# Set Boost options
+set(Boost_USE_STATIC_LIBS OFF)
+set(Boost_USE_MULTITHREADED OFF)
+set(Boost_USE_STATIC_RUNTIME OFF)
```

Comment:
We currently don't have any dependencies on boost, that is the buildfarm is not provisioned with it and I would love to avoid boost if possible.
Is there are way we can workaround it?

Reply:
Maybe that could help? https://stackoverflow.com/a/612112/4583130

Reply:
The reason I went with boost is actually that post... I'm not thrilled about using it either, but it was the only cross-platform way I could think of doing a directory search.

If it isn't too icky, I could try to implement the per-platform directory searches, and use a preprocessing directive to conditionally compile whichever one works for the OS that we're on...